src/generate/register.rs

```diff
@@ -210,12 +142,38 @@ pub fn render(
     }
 
     let mut out = vec![];
+    let methods = methods.iter().map(|s| format!("[`{0}`](crate::generic::Reg::{0})", s)).collect::<Vec<_>>();
+    let mut doc = format!("{}\n\nThis register you can {}. See [API](https://docs.rs/svd2rust/#read--modify--write-api).",
+                        &description, methods.join(", "));
+
+    if name_sc != "cfg" {
+        doc += format!("\n\nFor information about avaliable fields see [{0}]({0}) module", &name_sc).as_str();
+    }
     out.push(quote! {
-        #[doc = #description]
-        pub struct #name_pc {
-            register: vcell::VolatileCell<#rty>
-        }
+        #[doc = #doc]
+        pub type #name_pc = crate::Reg<#rty, #_name_pc>;
+
+        #[allow(missing_docs)]
+        #[doc(hidden)]
+        pub struct #_name_pc;
+    });
 
+    let doc = format!("`read()` method returns [{0}::R]({0}::R) reader structure", &name_sc);
+    if can_read {
+        out.push(quote! {
```

Comment:
Why not move the `let doc = ...` inside the `if` clause? Should be cheaper to compute and avoids the risk of accidentally using it in a wrong context.